### PR TITLE
MTDSA-14029: Add OAS YAML Files

### PR DESCRIPTION
**Pull request by tom-lloyd-hmrc**

### CODE REVIEW

File: resources/public/api/conf/1.0/endpoints/create_benefit.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          <tr>
+              <td><p>N/A - DEFAULT</p></td>
+              <td><p>Simulates a list of all state benefit types.</p></td>
+          </tr>
+          <tr>
+             <td><p>BENEFIT_TYPE_EXISTS</p></td>
+             <td><p>Simulates the scenario where a benefit of this type already exists.</p></td>
+          </tr>
+          <tr>
+              <td><p>NOT_FOUND</p></td>
```
Comment (SamuelJacobsHMRC) [inline resources/public/api/conf/1.0/endpoints/create_benefit.yaml:27]: Don't think this should be here as it is a create endpoint so there is nothing to be found. NOT_FOUND also isn't present in the spec.

File: resources/public/api/conf/1.0/endpoints/create_benefit.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+          RULE_END_DATE_BEFORE_START_DATE:
+            $ref: '../common/errors.yaml#/components/examples/ruleEndDateBeforeStartDate'
+          RULE_START_DATE_AFTER_TAX_YEAR_END:
+            $ref: '../common/errors.yaml#/components/examples/ruleStartDateAfterTaxYearEnd'
+          RULE_END_DATE_BEFORE_TAX_YEAR_START:
+            $ref: '../common/errors.yaml#/components/examples/ruleEndDateBeforeTaxYearStart'
+          RULE_INCORRECT_OR_EMPTY_BODY_SUBMITTED:
+            $ref: '../common/errors.yaml#/components/examples/ruleIncorrectOrEmptyBody'
+          RULE_BENEFIT_TYPE_EXISTS:
+            $ref: '../common/errors.yaml#/components/examples/ruleBenefitTypeExists'
```
Comment (SamuelJacobsHMRC) [inline resources/public/api/conf/1.0/endpoints/create_benefit.yaml:102]: Missing `ruleTaxYearNotSupported`

File: resources/public/api/conf/1.0/endpoints/create_benefit.yaml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    description: Forbidden
+    content:
+      application/json:
+        schema:
+          $ref: '../schemas/error_response.yaml'
+        examples:
+          CLIENT_OR_AGENT_NOT_AUTHORISED:
+            $ref: '../common/errors.yaml#/components/examples/clientOrAgentNotAuthorised'
+
+  "404":
```
Comment (SamuelJacobsHMRC) [inline resources/public/api/conf/1.0/endpoints/create_benefit.yaml:114]: Same as comment on the GTS - there is no NOT_FOUND error for this endpoint

File: resources/public/api/conf/1.0/endpoints/delete_benefit.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
+summary: Delete State Benefit
+description: |
+  This endpoint allows a developer to delete customer submitted state benefits.
```
Comment (SamuelJacobsHMRC) [inline resources/public/api/conf/1.0/endpoints/delete_benefit.yaml:3]: The description in the [spec](https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?pageId=207457230) is slightly different:
`
This endpoint allows a developer to delete customer submitted state benefits. If the customer attempts to delete a pre-populated state benefit then the operation will fail with a forbidden response.

A National Insurance number, tax year and benefit ID must be provided.
`

File: resources/public/api/conf/1.0/endpoints/amend_benefit.yaml [hunk 5]
```diff
@@ -0,0 +1,124 @@
+summary: Amend State Benefit
+description: |
+  This endpoint enables you to amend a previously created customer benefit, if the customer attempts to update
```
Comment (SamuelJacobsHMRC) [inline resources/public/api/conf/1.0/endpoints/amend_benefit.yaml:3]: The endpoint on the spec is slightly different:

`
This endpoint allows a developer to amend a previously created customer benefits. If the customer attempts to update a pre-populated benefit then this operation will fail with a forbidden response code.

A National Insurance number, tax year and benefit ID must be provided.
`

File: resources/public/api/conf/1.0/endpoints/amend_benefit.yaml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+          RULE_TAX_YEAR_NOT_SUPPORTED:
+            $ref: '../common/errors.yaml#/components/examples/ruleTaxYearNotSupported'
+          RULE_END_DATE_BEFORE_START_DATE:
+            $ref: '../common/errors.yaml#/components/examples/ruleEndDateBeforeStartDate'
+          RULE_START_DATE_AFTER_TAX_YEAR_END:
+            $ref: '../common/errors.yaml#/components/examples/ruleStartDateAfterTaxYearEnd'
+          RULE_END_DATE_BEFORE_TAX_YEAR_START:
+            $ref: '../common/errors.yaml#/components/examples/ruleEndDateBeforeTaxYearStart'
+          RULE_INCORRECT_OR_EMPTY_BODY_SUBMITTED:
+            $ref: '../common/errors.yaml#/components/examples/ruleIncorrectOrEmptyBody'
```
Comment (SamuelJacobsHMRC) [inline resources/public/api/conf/1.0/endpoints/amend_benefit.yaml:102]: Missing `RULE_TAX_YEAR_NOT_ENDED`

**Review by SamuelJacobsHMRC — CHANGES_REQUESTED** [review]

File: resources/public/api/conf/1.0/application.yaml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+          tokenUrl: https://api.service.hmrc.gov.uk/oauth/token
+          refreshUrl: https://api.service.hmrc.gov.uk/oauth/refresh
+          scopes:
+            write:self-assessment: Grant write access
+            read:self-assessment: Grant read access
+
+paths:
+  /individuals/state-benefits/{nino}/{taxYear}:
+    post:
+      $ref: "./endpoints/create_benefit.yaml"
```
[the remark this answers is not in the document]
Comment (tom-lloyd-hmrc) [reply resources/public/api/conf/1.0/application.yaml:51]: I didn't know about that convention, will rename.

File: resources/public/api/conf/1.0/common/pathParameters.yaml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+      required: true
+    benefitId:
+      name: benefitId
+      in: path
+      description: |
+        Unique identifier of the benefit type.
+        
+        Must conform to the regular expression
+        
+        `^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$`
```
[the remark this answers is not in the document]
Comment (tom-lloyd-hmrc) [reply resources/public/api/conf/1.0/common/pathParameters.yaml:28]: Oh that's much nicer!

File: resources/public/api/conf/1.0/endpoints/list_benefits.yaml [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+          <tr>
+            <td><p>N/A - DEFAULT</p></td>
+            <td><p>Simulates a list of all state benefit types.</p></td>
+          </tr>
+          <tr>
+              <td><p>N/A - DEFAULT<br>using <strong>benefitId</strong> parameter</p></td>
+              <td><p>Simulates a single HMRC held state benefit.</p></td>
+          </tr>
+          <tr>
+              <td><p>HMRC_AMENDED_AMOUNTS</p></td>
```
[the remark this answers is not in the document]
Comment (tom-lloyd-hmrc) [reply resources/public/api/conf/1.0/endpoints/list_benefits.yaml]: Asked about this in the TYS channel.

File: resources/public/api/conf/1.0/endpoints/delete_benefit_amounts.yaml [hunk 10]
```diff
@@ -0,0 +1,90 @@
+summary: Delete State Benefit Amounts
+description: |
+  This endpoint allows a developer to delete state benefit amount values supplied by the customer.
```
[the remark this answers is not in the document]
Comment (tom-lloyd-hmrc) [reply resources/public/api/conf/1.0/endpoints/delete_benefit_amounts.yaml]: I think the spec description might be wrong. Asking in the TYS channel.

**Review by SamuelJacobsHMRC — APPROVED** [review]

**Pull request merged**